#! /bin/sh -e

# DP: Revert ip6.arpa to ip6.int -> close BUG #119773
# DP: patch provided by Fabio "fabbione" M. Di Nitto <fabbione@fabbione.net>
# DP: XXX: This patch needs to be removed by the time we release a 2.3.x
# DP:      based dist!!!

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

diff -Naurd glibc-2.2.4.patched-ip6/resolv/gethnamaddr.c glibc-2.2.4/resolv/gethnamaddr.c
--- glibc-2.2.4.patched-ip6/resolv/gethnamaddr.c	Sun Dec  9 10:07:57 2001
+++ glibc-2.2.4/resolv/gethnamaddr.c	Sun Dec  9 10:13:53 2001
@@ -690,7 +690,7 @@
 				       uaddr[n] & 0xf,
 				       (uaddr[n] >> 4) & 0xf));
 		}
-		strcpy(qp, "ip6.arpa");
+		strcpy(qp, "ip6.int");
 		break;
 	default:
 		abort();
diff -Naurd glibc-2.2.4.patched-ip6/resolv/nss_dns/dns-host.c glibc-2.2.4/resolv/nss_dns/dns-host.c
--- glibc-2.2.4.patched-ip6/resolv/nss_dns/dns-host.c	Sun Dec  9 10:07:57 2001
+++ glibc-2.2.4/resolv/nss_dns/dns-host.c	Sun Dec  9 10:14:09 2001
@@ -282,7 +282,7 @@
       qp = qbuf;
       for (n = IN6ADDRSZ - 1; n >= 0; n--)
 	qp += sprintf (qp, "%x.%x.", uaddr[n] & 0xf, (uaddr[n] >> 4) & 0xf);
-      strcpy (qp, "ip6.arpa");
+      strcpy (qp, "ip6.int");
       break;
     default:
       /* Cannot happen.  */
